src/tests/compiler/clvm.rs

```diff
@@ -138,3 +150,66 @@ fn test_clvm_4() {
 
     assert!(result.equal_to(want[0].borrow()));
 }
+
+#[cfg(test)]
+fn does_number_need_extension_byte(n: Number) -> bool {
+    let mut iv = n.clone();
+    let eight_ones = 255_u32.to_bigint().unwrap();
+    while iv > eight_ones {
+        iv /= eight_ones.clone() + bi_one();
+    }
+    iv > 127_u32.to_bigint().unwrap()
+}
+
+// This seems like a reasonable way to unit test just the conversion functions
+// in a broad way i suppose.
+#[test]
+fn test_random_int_just_the_conversion_functions_and_no_other_things_from_the_stack_1() {
+    let mut rng = ChaChaRng::from_entropy();
+    for _ in 1..=200 {
+        let number_spec: RandomClvmNumber = rng.gen();
+
+        let to_bytes_clvm = bigint_to_bytes_clvm(&number_spec.intended_value).raw();
+        let to_bytes_unsigned = if number_spec.intended_value < bi_zero() {
+            None
+        } else {
+            Some(bigint_to_bytes_unsigned(&number_spec.intended_value).raw())
+        };
+
+        if number_spec.intended_value == bi_zero() {
```

Comment:
this case may not run. it's hard for me to judge how likely it is to run, but I would think we would always want to test this property.